go/vt/tabletserver/tx_engine.go

```diff
@@ -0,0 +1,252 @@
+// Copyright 2016, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tabletserver
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	log "github.com/golang/glog"
+
+	"github.com/youtube/vitess/go/timer"
+	"github.com/youtube/vitess/go/vt/concurrency"
+	"github.com/youtube/vitess/go/vt/dbconfigs"
+	"github.com/youtube/vitess/go/vt/vtgate/vtgateconn"
+)
+
+// TxEngine handles transactions.
+type TxEngine struct {
+	isOpen, twopcEnabled bool
+	coordinatorAddress   string
+	abandonAge           time.Duration
+	ticks                *timer.Timer
+
+	txPool *TxPool
+
+	// readPool is used for internal housekeeping work.
```

Comment:
It seems like the readPool is really only used by the 2pc engine / watchdog. Could it be moved to the twoPC object?